src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -250,18 +243,6 @@ export class AttachedAccountsTreeItem extends AzureParentTreeItem {
         return this._attachedAccounts;
     }
 
-    private async canConnectToLocalMongoDB(): Promise<boolean> {
```

Comment:
Another option is to keep this in, but add a timeout of like 1 second. You can leverage `Promise.race` for this

```typescript
private async canConnectToLocalMongoDB(): Promise<boolean> {
    async function timeout(): Promise<boolean> {
        await delay(1000);
        return false;
    }
    async function connect(): Promise<boolean> {
        try {
            const db: MongoClient | void = await connectToMongoClient(localMongoConnectionString, appendExtensionUserAgent());
            // grandfathered in
            // tslint:disable-next-line: no-floating-promises
            db.close();
            return true;
        } catch {
            return false;
        }
    }
    return await Promise.race([timeout(), connect()]);
}
```